core/station/impl/src/factories/blockchains/ethereum.rs

```diff
@@ -0,0 +1,295 @@
+                "test_recovered_address: {} {:?} {:?}",
+                hex::encode_prefixed(test_recovered_address).to_lowercase()
+                    == account_address.to_lowercase(),
+                hex::encode_prefixed(test_recovered_address),
+                account_address
+            ));
+        }
+
+        print(format!("signature: {:?}", signature));
+        let tx_signed = transaction.into_signed(signature);
+        let tx_envelope: alloy::consensus::TxEnvelope = tx_signed.into();
+        let tx_encoded = tx_envelope.encoded_2718();
+        print(format!("tx_encoded: {:?}", tx_encoded));
+
+        let sent_tx_hash = eth_send_raw_transaction(&self.chain, &tx_encoded).await;
+        print(format!("sent_tx_hash: {:?}", sent_tx_hash));
+
+        Ok(BlockchainTransactionSubmitted {
+            details: vec![(
+                TRANSACTION_SUBMITTED_DETAILS_TRANSACTION_HASH_KEY.to_owned(),
+                sent_tx_hash,
+            )],
+        })
+    }
+}
+
+async fn ecdsa_pubkey_of(account: &Account) -> Vec<u8> {
+    let (key,) = ecdsa_public_key(EcdsaPublicKeyArgument {
+        canister_id: None,
+        derivation_path: principal_to_derivation_path(&account),
+        key_id: get_key_id(),
+    })
+    .await
+    .expect("failed to get public key");
+    key.public_key
+}
+
+async fn get_address_from_account(account: &Account) -> String {
+    let public_key = ecdsa_pubkey_of(&account).await;
+    let address = get_address_from_public_key(&public_key);
+    hex::encode_prefixed(&address)
+}
+
+fn get_address_from_public_key(public_key: &[u8]) -> Address {
+    let verifying_key = ecdsa::VerifyingKey::from_sec1_bytes(&public_key)
+        .expect("Failed to create VerifyingKey from public key bytes");
+    alloy::signers::utils::public_key_to_address(&verifying_key)
+}
+
+fn get_key_id() -> EcdsaKeyId {
+    // TODO: check what we should use as a name
+    let name: String = "dfx_test_key".to_string();
+
+    EcdsaKeyId {
+        curve: EcdsaCurve::Secp256k1,
+        name,
+    }
+}
+
+fn principal_to_derivation_path(account: &Account) -> Vec<Vec<u8>> {
+    let account_principal = Principal::from_slice(&account.id);
+    const SCHEMA: u8 = 1;
+    vec![vec![SCHEMA], account_principal.as_slice().to_vec()]
+}
+
+pub async fn eth_send_raw_transaction(chain: &alloy_chains::Chain, raw_tx: &[u8]) -> String {
+    let config = None;
+    let services = get_evm_services(chain);
+
+    let cycles = 1000000000;
+
+    let raw_tx_hex = hex::encode_prefixed(raw_tx);
+    let send_result = EVM_RPC
+        .eth_send_raw_transaction(services.1, config, raw_tx_hex, cycles)
+        .await;
+    let status = match send_result {
+        Ok((res,)) => match res {
+            MultiSendRawTransactionResult::Consistent(status) => match status {
+                SendRawTransactionResult::Ok(status) => status,
+                SendRawTransactionResult::Err(e) => {
+                    ic_cdk::trap(format!("Error: {:?}", e).as_str());
```

Comment:
instead of directly using `ic_cdk::trap` we should instead use `crate::core::ic_cdk::trap` because that is prepared to run in tests